sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/AbstractDataDeserializer.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.schemaregistry.client.SchemaRegistryObject;
+import com.azure.data.schemaregistry.client.SchemaRegistryClient;
+import com.azure.data.schemaregistry.client.SchemaRegistryClientException;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Common implementation for all registry-based deserializers.
+ */
+public abstract class AbstractDataDeserializer extends AbstractDataSerDe {
+    private final ClientLogger logger = new ClientLogger(AbstractDataDeserializer.class);
+
+    protected final Map<String, ByteDecoder> byteDecoderMap = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor called by all concrete implementation constructors.
+     * Should only call parent constructor.
+     * @param schemaRegistryClient client to be used for fetching schemas by ID
+     */
+    protected AbstractDataDeserializer(SchemaRegistryClient schemaRegistryClient) {
+        super(schemaRegistryClient);
+    }
+
+    /**
+     * Special case constructor for Kafka deserializer's empty constructors.
+     */
+    protected AbstractDataDeserializer() { }
+
+    /**
+     * Fetches schema referenced by prefixed ID and deserializes the subsequent payload into Java object.
+     *
+     * @param payload byte payload, produced by an Azure Schema Registry client producer
+     * @return object, deserialized with the prefixed schema
+     * @throws SerializationException if deserialization of registry schema or message payload fails.
+     */
+    protected Object deserialize(byte[] payload) throws SerializationException {
+        if (payload == null) {
+            return null;
+        }
+
+        ByteBuffer buffer = ByteBuffer.wrap(payload);
+        String schemaGuid = getSchemaGuidFromPayload(buffer);
+        SchemaRegistryObject registryObject;
+        Object payloadSchema;
+
+        try {
+            registryObject = this.schemaRegistryClient.getSchemaByGuid(schemaGuid);
+            payloadSchema = registryObject.deserialize();
+        } catch (SchemaRegistryClientException e) {
+            throw logger.logExceptionAsError(
+                new SerializationException(String.format("Failed to retrieve schema for id %s", schemaGuid), e));
+        }
+
+        // TODO: how to handle unknown formats
```

Comment:
I think throwing is the correct choice here, if we can't determine the schema there isn't much we can do.

Reply:
Yeah the question here is whether or not we should offer a config to skip the messages where schema was not recognized.  Right now it will throw at the registry client level since there no schema parser will have been loaded.  I think we decided to punt the final decision after preview but I am leaning heavily towards throwing.

Reply:
Going to just take the comment out.